server/pbenchinacan/load_keycloak.sh

```diff
@@ -1,110 +1,98 @@
-#!/usr/bin/env -S bash -e
+#!/bin/bash -e
 
 # This scripts loads the keycloak server with following configuration
 # 1. A realm with the name pbench
-# 2. A client pbench-client under the realm pbench
-# 3. A user test_admin
+# 2. A client 'pbench-server' under the realm pbench
+# 3. A user test_user
 # 4. An ADMIN role under the client pbench-client
-# 5. ADMIN role assigned to the test_admin user.
+# 5. ADMIN role assigned to the test_user user.
 
 # At the moment the Pbench-server requires a Keycloak server with at least
-# no. 1 and no. 2 configuration specified above.
-# This scrip assumes that Keycloak server is running and does not already have
-# the configuration specified above.
+# a realm and a client configuration specified above.
+# This script assumes that Keycloak server is running and the master realm
+# username/password are admin/admin
 
 KEYCLOAK_HOST_PORT=${1:-"localhost:8080"}
+KEYCLOAK_REDIRECT_URI=${2:-"http://localhost:8080/*"}
+ADMIN_USERNAME="admin"
+ADMIN_PASSWORD="admin"
+
 echo
-echo "KEYCLOAK_HOST_PORT: $KEYCLOAK_HOST_PORT"
+echo "KEYCLOAK_HOST_PORT: ${KEYCLOAK_HOST_PORT}"
 
 echo
 echo "Getting admin access token"
 echo "--------------------------"
 
-ADMIN_TOKEN=$(curl -s -X POST "http://$KEYCLOAK_HOST_PORT/realms/master/protocol/openid-connect/token" \
+ADMIN_TOKEN=$(curl -s -X POST "http://${KEYCLOAK_HOST_PORT}/realms/master/protocol/openid-connect/token" \
   -H "Content-Type: application/x-www-form-urlencoded" \
-  -d "username=admin" \
-  -d 'password=admin' \
+  -d "username=${ADMIN_USERNAME}" \
+  -d "password=${ADMIN_PASSWORD}" \
   -d 'grant_type=password' \
   -d 'client_id=admin-cli' | jq -r '.access_token')
 
-echo "ADMIN_TOKEN=$ADMIN_TOKEN"
+echo "ADMIN_TOKEN=${ADMIN_TOKEN}"
 echo
 
 echo "Creating realm"
 echo "--------------"
 
-curl -i -X POST "http://$KEYCLOAK_HOST_PORT/admin/realms" \
-  -H "Authorization: Bearer $ADMIN_TOKEN" \
+curl -i -X POST "http://${KEYCLOAK_HOST_PORT}/admin/realms" \
+  -H "Authorization: Bearer ${ADMIN_TOKEN}" \
   -H "Content-Type: application/json" \
   -d '{"realm": "pbench", "enabled": true}'
 
 echo "Creating client"
 echo "---------------"
 
-CLIENT_CONF=$(curl -si -X POST "http://$KEYCLOAK_HOST_PORT/admin/realms/pbench/clients" \
-  -H "Authorization: Bearer $ADMIN_TOKEN" \
+CLIENT_CONF=$(curl -si -X POST "http://${KEYCLOAK_HOST_PORT}/admin/realms/pbench/clients" \
+  -H "Authorization: Bearer ${ADMIN_TOKEN}" \
   -H "Content-Type: application/json" \
-  -d '{"clientId": "pbench-client", "directAccessGrantsEnabled": true, "serviceAccountsEnabled": true, "redirectUris": ["http://localhost:8080/*"]}')
+  -d '{"clientId": "pbench-client", "directAccessGrantsEnabled": true, "serviceAccountsEnabled": true, "redirectUris": "'"${KEYCLOAK_REDIRECT_URI}"'"]}')
```

Comment:
I think the value of `"redirectUris"` is messed up.  Shouldn't it be:
```
  -d '{"clientId": "pbench-client", "directAccessGrantsEnabled": true, "serviceAccountsEnabled": true, "redirectUris": ["'${KEYCLOAK_REDIRECT_URI}'"]}')
```
That is, I added the missing opening `[` as well as fixing the quotes.